pkg/engine/engine.go

```diff
@@ -233,6 +234,21 @@ func (e *Engine) BytesScanned() uint64 {
 	return e.bytesScanned
 }
 
+func (e *Engine) dedupeAndSend(chunkResults []detectors.ResultWithMetadata) {
+	dedupeMap := make(map[string]struct{})
+	for _, result := range chunkResults {
+		// dedupe by comparing the detector type, raw result, and source metadata
+		// NOTE: in order for the PLAIN decoder to maintain precedence, make sure UTF8 is the first decoder in the
+		// default decoders list
+		if _, ok := dedupeMap[result.DetectorType.String()+string(result.Raw)+fmt.Sprintf("%+v", result.SourceMetadata)]; ok {
+			continue
+		}
+		dedupeMap[result.DetectorType.String()+string(result.Raw)+fmt.Sprintf("%+v", result.SourceMetadata)] = struct{}{}
```

Comment:
**Nit (optional):** A single `Sprintf` would be a little easier to read imo:

```go
key := fmt.Sprintf("%s%s%+v", result.DetectorType.String(), string(result.Raw), result.SourceMetadata)
if _, ok := dedupeMap[key]; ok {
    continue
}
dedupeMap[key] = struct{}{}

// ...
```

Also, should we be using `RawV2` instead? or both?

Reply:
probably both. for the tests cases that I was testing with rawv2 was empty string